src/NHSD.GPIT.BuyingCatalogue.UI.Components/Views/Shared/TagHelpers/Pagination/PageNumberPaginationTagHelper.cs

```
@@ -111,7 +111,13 @@ private string GenerateNextLinkHref()
 
         private string GenerateLinkHrefForPage(int pageNumber)
         {
-            var builder = new UriBuilder(ViewContext.HttpContext.Request.GetEncodedUrl())
+            var domain = Environment.GetEnvironmentVariable("DOMAIN_NAME");
+
+            var url = !string.IsNullOrEmpty(domain)
+                ? $"{ViewContext.HttpContext.Request.Scheme}://{domain}{ViewContext.HttpContext.Request.Path}"
```

You could use `UriBuilder` here to construct the URL as it would fail faster if there are any issues with the URL parts.

---

Changed to use relative path